src/MICore/Debugger.cs

```diff
@@ -1118,11 +1114,34 @@ private void HandleThreadGroupStarted(Results results)
             {
                 lock (_debuggeePids)
                 {
-                    _debuggeePids.Add(idString, pid);
+                    _debuggeePids.Add(threadGroupId, pid);
                 }
             }
         }
 
+        private void HandleThreadGroupExited(Results results)
+        {
+            string threadGroupId = results.TryFindString("id");
+
+            bool shouldExit = false;
+
+            if (!String.IsNullOrEmpty(threadGroupId))
+            {
+                lock (_debuggeePids)
+                {
+                    if (_debuggeePids.Remove(threadGroupId))
+                    {
+                        shouldExit = _debuggeePids.Count == 0;
+                    }
+                }
+            }
+
+            if (shouldExit)
+            {
+                OnDebuggerProcessExit(null);
```

Comment:
So this is the code path that acts like gdb just disappeared. I think this probably isn't want you want. It seems like we should handle it like:

https://github.com/Microsoft/MIEngine/blob/master/src/MICore/Debugger.cs#L194

If that doesn't work for some reason, we likely need to plumb something else along to OnDebuggerProcessExit to make sure we don't report telemetry that GDB exited.